src/odemis/driver/xt_client.py

```diff
@@ -1938,7 +1989,7 @@ def applyAutofocus(self, detector):
         f._autofocus_lock = threading.Lock()
         f._must_stop = threading.Event()  # cancel of the current future requested
         f.task_canceller = self._cancelAutoFocus
-        f._channel_name = DETECTOR2CHANNELNAME[detector]
+        f._channel_name = self._scanner.channel
```

Comment:
I don't see a `._scanner` attribute. That should be something like `self.parent._scanner.channel`, right?
I think the focus is hard coded to be for the ebeam in the xtlib API. So I'd also recommend to check at init that there is a _scanner (ie, e-beam "scanner" or "mb-scanner").